README.md

```diff
@@ -1,5 +1,95 @@
-#  Under construction
+## Motivation
 
-Currently migrating code from other sources.
+Many fields manipulate collections of Tensors of different shapes. For example, paragraphs of text, images of different sizes or audio files of different lengths. We don't have a first class abstraction that enables the concurrent manipulation of collections of this type of data. We further often need to batch arbitrary data and operations for efficiency.
 
-![alt text](assets/sign.png "Construction sign")
+## Decription
+
+NestedTensor is an abstraction of torch tensor which allows working with data of different sizes and length. 
+
+In general, there are two cases for which NestedTensors provide computational representations.
+
+**Lists of Tensors**
+Each Tensor constituent of the list it represents, if any, must be of its dtype, layout and device. The dimension of a constituent Tensor must be one less than the dimension of the NestedTensor. An empty list of Tensors yields a NestedTensor of dimension one.
+
+**Lists of NestedTensors**
+Each constituent NestedTensor must be of its dtype, layout and device. The dimension of a constituent NestedTensor must be one less than the dimension of the NestedTensor.
+
+## Prerequisites
+
+You will need this packages to be installed:
+
```

Comment:
Qualify what each package is needed for. The raw repo only requires pytorch, the example notebooks require torchvision etc.